src/routers/xcm/Router.sol

```diff
@@ -44,6 +95,19 @@ contract ConnectorXCMRouter {
         _;
     }
 
+    function file(bytes32 what, uint64 buyExecutionWeightLimit, uint64 transactWeightAtMost, uint256 feeAmount)
+        external
+        auth
+    {
+        if (what == "xcmWeightInfo") {
+            xcmWeightInfo = XcmWeightInfo(buyExecutionWeightLimit, transactWeightAtMost, feeAmount);
+        } else {
+            revert("CentrifugeXCMRouter/file-unrecognized-param");
+        }
+
+        emit File(what, xcmWeightInfo);
+    }
+
     function handle(bytes memory _message) external onlyCentrifugeChainOrigin {
```

Comment:
is `centrifugeChainOrigin` settable somewhere or are we locked once deployed?

Reply:
yeah `centrifugeChainOrigin` is set in the `constructor` and it is immutable once set. This is fine tho because that value is derived from the centrifuge parachain Id so it's not going to change within the same network